specification.md

```diff
@@ -0,0 +1,344 @@
+and verified by the recipient before deserializing. A backwards compatible
+variant is available.
+
+# Specification
+
+## Current format
+
+The
+[current signature format](https://github.com/in-toto/docs/blob/master/in-toto-spec.md#42-file-formats-general-principles)
+has a BODY that is a regular JSON object and a signature over the
+[Canonical JSON] serialization of BODY.
+
+```json
+{
+  "signed": <BODY>,
+  "signatures": [{
+    …,
+    "sig": "<Hex(Sign(CanonicalJson(BODY)))>"
+  }, …]
+}
+```
+
+To verify, the consumer parses the whole JSON file, re-serializes BODY using
+Canonical JSON, then verifies the signature.
+
+## Proposed format
+
+Instead of relying on Canonical JSON, the producer records the signed bytes
+exactly as signed and the consumer verifies those exact bytes before parsing. In
+addition, the signature now includes an authenticated `payloadType` field
+indicating how to interpret the payload.
+
+```json
+{
+  "payload": "<Base64(SERIALIZED_BODY)>",
+  "payloadType": "<PAYLOAD_TYPE>",
+  "signatures": [{
+    …,
+    "sig": "<Base64(Sign(PAE([PAYLOAD_TYPE, SERIALIZED_BODY])))>"
+  }, …]
+}
+```
+
+where PAE is the
+[PASETO Pre-Authentication Encoding](https://github.com/paragonie/paseto/blob/master/docs/01-Protocol-Versions/Common.md#authentication-padding):
+
+```none
+PAE([type, body]) := le64(2) || le64(len(type)) || type || le64(len(body)) || body
+le64(n) := 64-bit little-endian encoding of `n`, where 0 <= n < 2^63
+```
+
+The PAYLOAD_TYPE is a URI indicating how to interpret SERIALIZED_BODY. It
+encompasses the content type (JSON, CBOR, etc.), the purpose, and the schema
+version of the payload. This obviates the need for the `_type` field within
+in-toto/TUF payloads. Examples:
+
+-   https://in-toto.to/Link/v0.9
+-   https://in-toto.to/Layout/v0.9
+-   https://theupdateframework.com/Root/v1.0.5
+-   etc...
+
+The switch from Hex to Base64 for `sig` is to save space and to be consistent
+with `payload`.
+
+## Steps
+
+To sign:
+
+-   Serialize BODY according to PAYLOAD_TYPE. Call the result SERIALIZED_BODY.
+-   Sign PAE([PAYLOAD_TYPE, SERIALIZED_BODY]), base64-encode the result, and
+    store it in `sig`.
+-   Base64-encode SERIALIZED_BODY and store it in `payload`.
+-   Store PAYLOAD_TYPE in `payloadType`.
+
+To verify:
+
+-   Base64-decode `payload`; call this SERIALIZED_BODY. Reject if the decoding
+    fails.
+-   Base64-decode `sig` and verify PAE([PAYLOAD_TYPE, SERIALIZED_BODY]). Reject
+    if either the decoding or the signature verification fails.
+-   Parse SERIALIZED_BODY according to PAYLOAD_TYPE. Reject if the parsing
+    fails.
+
+Either standard or URL-safe base64 encodings are allowed. Signers may use
+either, and verifiers must accept either.
+
+## Backwards compatible signatures
+
+To convert existing signatures from the current format to the new format, the
+special `"backwards-compatible-json"` payload type indicates that the signature
```

Comment:
Might as well put `"backwards-compatible-json"` in a URI to be consistent.

Reply:
I've updated it to recommend appending it to the regular URI for the document. Does that make sense?